### PR TITLE
Retry Service Dependencies on Startup -- ExportClient

### PR DESCRIPTION
**Pull request by brandonforster**
Issue #441 

Also included in this pull request are commits normalizing the spelling of the word "Microservice" in documentation and logs to console.

### CODE REVIEW

**Comment by tsconn23** [comment]
recheck

**Comment by tsconn23** [comment]
@brandonforster Please investigate broken build related to docker image
```
23:09:05 [0m[91mmake: *** [Makefile:43: cmd/export-client/export-client] Error 2
23:09:05 [0mThe command '/bin/sh -c make cmd/export-client/export-client' returned a non-zero code: 2
23:09:08 Makefile:108: recipe for target 'docker_export_client' failed
23:09:08 make: *** [docker_export_client] Error 2
23:09:08 Build step 'Execute shell' marked build as failure
```

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,7 @@ cmd/core-command/core-command:
 	$(GO) build $(GOFLAGS) -o $@ ./cmd/core-command
 
 cmd/export-client/export-client:
-	$(GO) build $(GOFLAGS) -o $@ ./cmd/export-client
+	$(GOCGO) build $(GOFLAGS) -o $@ ./cmd/export-client
```
Comment (jduranf) [inline Makefile]: Why this change?

Comment (brandonforster) [reply Makefile]: Something in this change set introduced a dependency somewhere in export clients tree on ZeroMQ, which requires that CGO compilation is enabled.

Comment (tsconn23) [reply Makefile]: @brandonforster I am able to build this PR without the above change to the makefile. Please verify locally. If you are still having issues, let me know. We don't want to introduce CGo unnecessarily.
```
~/go/src/github.com/edgexfoundry/edgex-go $ git branch
  california

  master
* pr_558
[tconn@Borg9 14:02]
~/go/src/github.com/edgexfoundry/edgex-go $ make build
CGO_ENABLED=0 go build -ldflags "-X github.com/edgexfoundry/edgex-go.Version=0.7.0" -o cmd/config-seed/config-seed ./cmd/config-seed
CGO_ENABLED=0 go build -ldflags "-X github.com/edgexfoundry/edgex-go.Version=0.7.0" -o cmd/export-client/export-client ./cmd/export-client
CGO_ENABLED=1 go build -ldflags "-X github.com/edgexfoundry/edgex-go.Version=0.7.0" -o cmd/export-distro/export-distro ./cmd/export-distro
```

File: internal/export/client/init.go [hunk 2]
```diff
@@ -8,88 +8,111 @@ package client
...
 const (
 	PingApiPath = "/api/v1/ping"
 )
 
 // Global variables
-var dbc export.DBClient
+var dbClient export.DBClient
 var LoggingClient logger.LoggingClient
+var Configuration *ConfigurationStruct
+var chEvents chan interface{} //A channel for "domain events" sourced from event operations
```
Comment (tsconn23) [inline internal/export/client/init.go]: This variable isn't used anywhere in this service and can be removed.

File: internal/export/client/init.go [hunk 3]
```diff
@@ -102,10 +125,50 @@ func newDBClient(dbType string, config db.Configuration) (export.DBClient, error
...
+		CheckInterval:  conf.CheckInterval,
+		ConsulAddress:  conf.ConsulHost,
+		ConsulPort:     conf.ConsulPort,
+	})
+
+	if err != nil {
+		return fmt.Errorf("connection to Consul could not be made: %v", err.Error())
+	} else {
+		// Update configuration data from Consul
+		if err := consulclient.CheckKeyValuePairs(&conf, internal.ExportClientServiceKey, strings.Split(conf.ConsulProfilesActive, ";")); err != nil {
```
Comment (tsconn23) [inline internal/export/client/init.go]: There is an error on this line that causes a panic when the service is started with the "--consul" parameter. Compare with core/data/init.go.
